### PR TITLE
Fix: context for afterFind

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description
Adds context to `Parse.Cloud.afterFind`

Related issue: #7058. Does not close this issue, but works towards.

### Approach
Add context to `Parse.Cloud.afterFind`

### CODE REVIEW

File: spec/CloudCode.spec.js [hunk 1]
```diff
@@ -3157,4 +3157,14 @@ describe('afterLogin hook', () => {
 
     await Parse.Cloud.run('contextTest', {}, { context: { a: 'a' } });
   });
+
+  it('afterFind should have access to context', async () => {
+    Parse.Cloud.afterFind('TestObject', req => {
+      expect(req.context.a).toEqual('a');
```
Comment (mtrezza) [inline spec/CloudCode.spec.js:3163]: Is the context accessible anywhere after `afterFind`, or is that the last stage for the context object?

Comment (dblythy) [reply spec/CloudCode.spec.js:3163]: Not 100% sure but I think that's the last stage. `this.context` will obvs still be available on RestQuery but it's not referenced anywhere else, and after `maybeRunAfterFind`, `RestQuery` returns.

Comment (mtrezza) [reply spec/CloudCode.spec.js:3163]: Okay, just wanted to make sure whether the test case should also modify the context and test whether the change permeates correctly. But that doesn't apply to the last stage.

**Review by mtrezza — CHANGES_REQUESTED** [review]
Can you please also add the changelog entry? Then it should be good for merge.

File: spec/CloudCode.spec.js (hunk 1 above)
Comment (dblythy) [reply spec/CloudCode.spec.js:3163]: Do you mean if context is mutated in beforeFind, make sure the mutations show in context on afterFind?

Comment (mtrezza) [reply spec/CloudCode.spec.js:3163]: I meant that the mutation made in afterFind shows in context at the next stage, but apparently there is no next stage, so we're good here. 🙂

**Review by mtrezza — APPROVED** [review]
Looks good to me! Thanks for this PR 🙂

@dblythy I forgot the changelog entry, can you please add that and this should be ready to fly.

**Review by mtrezza — CHANGES_REQUESTED** [review]
Can you merge master into this branch to resolve the merge conflict? Let me know if you need help with this.

**Review by mtrezza — APPROVED** [review]

**Review by Moumouls — APPROVED** [review]

**Comment by dplewis** [comment]
@matheusfrozzi I think you may have found a bug. I see a test for beforeFind and afterFind but not both together. Is this something you would like to work on? Can you open a new issue so we may track and investigate?

**Comment by matheusfrozzi** [comment]
I'm taking a look at the code right now, trying to find how to make it work.
I will open an issue.

I found in the ParseHoock.spec it:
  it('should not set context on beforeFind', () => {

Saying that I'm not allowed to do what I'm trying to do. So I don't know if it was supposed to work.

**Comment by matheusfrozzi** [comment]
Hey guys, I know I couldn't work on this one, but I think that fixing this issue of adding context in beforeFind that is not available on afterFind would be good before the new main release.

**Comment by mtrezza** [comment]
@matheusfrozzi Would you please open a new issue for this so that it’s on our radar? An additional PR with at least a failing test would be great, but even just the issue helps.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0](https://github.com/parse-community/parse-server/releases/tag/5.0.0)

**Pull request merged**